build-selftests/build_selftests.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -euo pipefail
+
+THISDIR="$(cd $(dirname $0) && pwd)"
+
+source "${THISDIR}"/../helpers.sh
+
+VMLINUX_BTF="$1"
+KERNEL="$2"
+TOOLCHAIN="$3"
+TOOLCHAIN_NAME="$(echo $TOOLCHAIN | cut -d '-' -f 1)"
```

Comment:
can we move lines upto line 20 to some kind of a helper shell scrip that will determine tool chain? it seems these are repeated in all 3 shell files

Reply:
I'd think that arguments to the script should almost certainly be assigned in the script itself and not some sourced helper, otherwise we most likely end up obfuscating what the script expects as input. A similar argument goes for options setting the behavior of the shell and the sourcing of the helper script, in my opinion. That should be done in the top level script itself, or we just end up with non-obvious state changes.

For the setting of `LLVM_VER` I tend to agree, but only insofar as that could be a function, not that the variable setting should happen in a different file. That being said, I'd think that should be done as part of a separate change. Except for the samples action, the scripts are just copied and pasted from the existing source and I don't think we should modify them in significant ways while we copy them around. So this really is an issue in existing code. What do you think?

Reply:
yeah, I was talking about TOOLCHAIN_NAME and VERSION parsing. we can do it as a separate change, fine with me. it is just we have it the same in all 3 files which makes me worried in case we update any of these

Reply:
Sounds good. I will create a follow up change to factor this logic out.